### DISCUSSION TITLE
MTEX : Grain boundary approximation by line

### DISCUSSION THREAD

**Discussion by kavege** [mtex-toolbox/mtex#1492] [Ask Anything]
Good morning,
I would like to approximate with MTEX all grains boundaries with lines. but for that i would like to have access to the grains boundaries id, something which is not impossible with MTEX.
has somebody already done it or could help me ?

thank you.

**Comment by kilir** [comment]
Hi,
there's no id, just an index. However, if you want to access coordinates of vertices, it's roughly like that:

```
V = grains.V;
poly = grains.poly;

for ig = 1:length(grains)
  
  Vg = V(poly{ig},:);
....
```
Cheers,
Rüdiger

**Reply by kavege** [reply]
Hi,
Ok. but how could i know that it's the first or the last vertex. Because i want to know for each interface between two grains what are the coordinates of the first and the last vertices in order to compute my line.
best regards,
romeo

**Reply by kilir** [reply]
Hi,
if I understand correctly, you want to approximate all segments of a contact between to grains by just one line defined by the most distant vertices? Wouldn't that be just the interconnections between triple points?
Cheers,
Rüdiger

**Reply by kavege** [reply]
Yes it's that. I can do it with triple junctions. But for that,  I need to know for a given Triple junctions, the ids (and the coordinates) of the 3 closest triple junctions. I check grains.triplePoints.nextVertexId but it's not what i want.

**Reply by kilir** [reply]
Triple points have the properties x and y, possibly doing a Delauny triangulation on those and discarding those edges which should not dissect a grain (= same grainId on both sides) might be an idea.
Cheers,
Rüdiger

**Comment by zmichels** [comment]
I haven’t used the PolyLX toolbox in a while, but if I recall correctly, it once was possible to convert an MTEX grainset/boundaries to a shapefile for export (.shp). I might be wrong. And that was a while ago (maybe issues with compatibility now?)… also… it requires a separate toolbox even if it did work still.
